flang/lib/Semantics/resolve-names.cpp

```diff
@@ -1639,6 +1641,14 @@ bool OmpVisitor::Pre(const parser::OpenMPDeclareMapperConstruct &x) {
   return false;
 }
 
+bool OmpVisitor::Pre(const parser::OmpMapClause &x) {
+  const auto &mid{std::get<parser::OmpMapperIdentifier>(x.t)};
+  if (const auto &mapperName{mid.v})
+    mapperName->symbol =
+        &MakeSymbol(*mapperName, MiscDetails{MiscDetails::Kind::ConstructName});
```

Comment:
IIUC, the MAPPER modifier should refer to a preexisting identifier, so this should be a name lookup, not creating a new symbol.  The DECLARE MAPPER should declare the identifier (and create the symbol).  If the lookup fails, you could create a dummy, and report it as an error later on.